pkg/planner/cardinality/selectivity.go

```diff
@@ -820,10 +898,11 @@ func getEqualCondSelectivity(sctx sessionctx.Context, coll *statistics.HistColl,
 	}
 	val := types.NewBytesDatum(bytes)
 	if outOfRangeOnIndex(idx, val) {
+		realtimeCnt, _ := coll.GetScaledRealtimeAndModifyCnt(idx)
```

Comment:
I tend to leave this unchanged for now.
Because this will only happen for the mv index, and not all places will need this scaling for the mv index. Indeed, it needs some consideration to choose the correct row count to use. But I believe there won't be more new functionality that needs this logic for some time. So I tend not to make this logic more generalized.